### PR TITLE
Change to CGMES import of SVC regulation

### PR DESCRIPTION
**Pull request by MioRtia**
Signed-off-by: RALAMBOTIANA MIORA <miora.ralambotiana@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?**
Bug fix?

**What is the current behavior?** *(You can also link to an open issue here)*
When the SVC's regulating control is disabled, the SVC does not regulate even if its `controlEnabled` property is set to true.

**What is the new behavior (if this is a feature change)?**
When the SVC's regulating control is disabled and its `controlEnabled` property is set to true, the SVC regulates with local default regulation.

**Does this PR introduce a breaking change or deprecate an API?**
No

### CODE REVIEW

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/RegulatingControlMappingForStaticVarCompensators.java [hunk 1]
```diff
@@ -103,6 +97,12 @@ private boolean setRegulatingControl(RegulatingControl control, StaticVarCompens
         StaticVarCompensator.RegulationMode regulationMode;
 
         boolean okSet = false;
+        if (!control.enabled && rc.controlEnabledProperty) {
+            context.fixed("SVCControlEnabledStatus", String.format("Regulating control of %s is disabled but controlEnabled property is set to true." +
```
Comment (annetill) [inline cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/RegulatingControlMappingForStaticVarCompensators.java]: Maybe you should add in the comment that it is the property of the equipment.

**Comment by annetill** [comment]
Thanks for your detailed answer. I understand that this PR only deals with SVC and nothing has to be changed in the CGMES conversion for generators and tap changers in case of having ````controlEnabled```` for the equipment  ````regulatingControl```` disabled.

For SVC, I understand that, in case of a disabled/wrong regulating control:
* The IIDM regulation status is given by the SSH profile ````cim:RegulatingCondEq.controlEnabled```` ;
** In case of a ````true```` status, we take the mode from the ````EQ```` profile using field ````StaticVarCompensator.sVCControlMode```` ;
*** In case of `````voltage````, we take the set point from the ````EQ```` profile field ````voltageSetPoint`````;
*** In case of `````reactivePower````, we take the set point from the ````SSH```` profile field ````StaticVarCompensator.q`````;

Is it what you mean ? If yes, we will modify this PR in consequence.

Thanks for your review !

**Comment by MioRtia** [comment]
Thank you both for your inputs @annetill @zamarrenolm! 

> CGMES documentation is not clear about the priority to use regulation control or the attributes defined at equipment level in Static Var Compensators. My suggestion would be regulation control should be considered first. If disabled or no info available, then we should rely on sVCControlMode and voltageSetPoint or q attributes, depending if control mode is voltage or reactivePower

Isn't it what is done in this PR? The regulation from EQ is only activated when the regulating control is missing or disabled **and** controlEnabled is set to true. Or is there something I am missing?

**Comment by annetill** [comment]
Do you rely on field ````SVCControlMode```` from EQ profile ? If yes, just wait for Luma's answer.

**Comment by MioRtia** [comment]
> Do you rely on field `SVCControlMode` from EQ profile ? If yes, just wait for Luma's answer.

Yes, I do, this is checked in the `setDefaultRegulatingControl` method.

**Comment by zamarrenolm** [comment]
> > Do you rely on field `SVCControlMode` from EQ profile ? If yes, just wait for Luma's answer.
> 
> Yes, I do, this is checked in the `setDefaultRegulatingControl` method.

Yes you are right, I've reviewed the code and follows my recommendation. No additional changes are required.

Only additional thing is that I am not sure that `controlEnabled` property should be checked for `true` when we are setting the default control data, as it is an attribute directly related to Regulating Controls. Even if `controlEnabled` is `false` (it refers to its regulation control) we should use and apply info from equipment definition.

What do you think?

**Pull request merged**